sdk-lib/src/main/java/cash/z/ecc/android/sdk/model/Zatoshi.kt

```diff
@@ -0,0 +1,31 @@
+package cash.z.ecc.android.sdk.model
+
+/**
+ * A unit of currency used throughout the SDK.
+ *
+ * End users (e.g. app users) generally are not shown Zatoshi values.  Instead they are presented
+ * with ZEC, which is a decimal value represented only as a String.  ZEC are not used internally,
+ * to avoid floating point imprecision.
+ */
+data class Zatoshi(val value: Long) {
+    init {
+        require(value >= MIN_INCLUSIVE) { "Zatoshi must be in the range [0, $MAX_INCLUSIVE]" }
```

Comment:
No big deal, but I'd suggest to change the failure message to _"Zatoshi must be in the range [$MIN_INCLUSIVE, $MAX_INCLUSIVE]"_.